src/app/component/dashboard/administrator/registered-clinics/registered-clinic-detail/registered-clinic-detail.component.ts

```diff
@@ -0,0 +1,138 @@
+
+  @ViewChild("clinic") clinic: Modal;
+
+  data: any;
+  clinicId: number;
+  language: any;
+  numberOfEmployees: number = 0;
+  loading = true;
+  theme: string;
+  showDialog: boolean = false;
+  isFormDirty: boolean = false;
+  imagePath: any;
+
+  constructor(
+    public accountService: AccountService,
+    public helpService: HelpService,
+    public router: ActivatedRoute,
+    public location: Location,
+    public sanitizer: DomSanitizer,
+  ) { }
+
+  ngOnInit() {
+    this.clinicId = +this.router.snapshot.paramMap.get('id');
+    this.language = this.helpService.getLanguage();
+    this.getSuperAdmin();
+    this.getClinic();
+    
+    setTimeout(() => {
+      this.modelData();
+    }, 1000);
+  }
+
+  modelData() {
+    this.data.birthday = new Date(this.data.birthday);
+    this.data.incompanysince = new Date(this.data.incompanysince);
+    // if (this.data.type === 1) {
+    //   this.selectedValue = "Admin";
+    // } else if (this.data.type === 2) {
+    //   this.selectedValue = "Manager";
+    // } else if (this.data.type === 6) {
+    //   this.selectedValue = "Read only scheduler";
+    // } else {
+    //   this.selectedValue = "Employee";
+    // }
+    this.loading = false;
+  }
+
+  getSuperAdmin() {
+    this.accountService.getSuperadmin(this.clinicId).subscribe(res => {
+      this.data = res[0];
+      if (this.data.img && this.data.img.data.length !== 0) {
+        const TYPED_ARRAY = new Uint8Array(this.data.img.data);
+        const STRING_CHAR = String.fromCharCode.apply(null, TYPED_ARRAY);
+        let base64String = btoa(STRING_CHAR);
+        let path = this.sanitizer.bypassSecurityTrustUrl(
+          "data:image/png;base64," + base64String
+        );
+      
+        this.imagePath = path;
+      } else {
+          this.imagePath = "../../../../../assets/images/users/defaultUser.png";
+          console.log('? imagepathg ', this.imagePath);
+      }
+    });
+  }
+
+  getClinic() {
+    this.accountService.getClinicEmployees(this.clinicId).subscribe((res: any) => {
+      this.numberOfEmployees = res.length;
+    });
+  }
+
+  editOptions() {
+    // this.workTimeData();
+    this.clinic.closeOnEscape = false;
+    this.clinic.closeOnOutsideClick = false;
+    this.clinic.hideCloseButton = true;
+    this.clinic.open();
+    // this.changeTheme(this.theme);
+  }
+
+  confirmClose(): void {
+    this.clinic.modalRoot.nativeElement.focus();
+    if(this.isFormDirty) {
+      this.showDialog = true;
+    }else {
+      this.clinic.close()
+      this.showDialog = false;
+      this.isFormDirty = false
+    }
+  }
+
+  receiveConfirm(event: boolean): void {
+    if(event) {
+      this.clinic.close();
+      this.isFormDirty = false;
+    }
+      this.showDialog = false;
+  }
+
+  isDirty(): void {
+    this.isFormDirty = true;
+  }
+
+  updateClinic() {
+    this.accountService.updateRegisteredClinic(this.data).subscribe((data) => {
+      if (data) {
+        Swal.fire({
+          title: "Successfull!",
```

Comment:
This message need to be translated.

Reply:
We have for this generic method - successToastr in helpService. Just need to send parameters like title and text.